thunder/__init__.py

```diff
@@ -474,7 +474,12 @@ def apply_transforms_and_build_cache_entry(cd, cs, cache_info, prologue_trc, com
             computation_traces.append(computation_trc)
 
             if not compile_options.get("skip_inplace_alias_updates", True):
-                computation_traces.append(insert_alias_updates(computation_trc))
+                alias_tensor_indices = []
+                if alias_tensor_indices_str := cache_info["alias_tensor_indices"]:
+                    alias_tensor_indices: list[list[int]] = [
+                        [int(i) for i in s.split(",")] for s in alias_tensor_indices_str.split("-")
+                    ]
```

Comment:
Can we prepare `alias_tensor_indices_str` when inplace_functionalization or inplace_alias_updates are enabled?

This is a bit orthogonal to this pull request, but I suppose we should exclude `skip_inplace_functionalization=False` and `skip_inplace_alias_updates=False`.
So in summary I think what I find nice is something like

```python
            utils.check(
                skip_inplace_functionalization or skip_inplace_alias_updates,
                lambda: "We cannot use both `alias_updates` and `functionalization` for in-place ops",
            )
            alias_tensor_indices: list[list[int]] = ...
```

before the two paths of in-place handling

Reply:
My understanding is that the two strategies are to be blended together eventually.  And since the alias update strategy is still experimental, I think I don't want to put in such a check just yet.